### ISSUE TITLE
support illumos systems

### ISSUE THREAD

**Issue by sjorge** [jclulow/vmware-sercons#2]
I'm not a C programmer so I have no idea if these patches make sens or are even harmful.
But the following makes it compile (and work as far as I can tell) on SmartOS! (assuming illumos in general)

The Makefile is missing `-lsocket and -lnsl` on illumos.

``` patch
--- vmware-sercons/Makefile     2016-10-09 13:46:48.461061921 +0200
+++ vmware-sercons_/Makefile    2016-10-09 13:44:42.606839225 +0200
@@ -6,6 +6,9 @@
 CFLAGS += -m32
 CFLAGS += -mmacosx-version-min=10.6
 endif
+ifeq ($(OS), SunOS)
+CFLAGS += -lsocket -lnsl
+endif

 PROG = sercons
```

sun_len does not appear to be a generic unix interface, illumos does not support.
I found some stuff on BSD mailing lists to just drop it and use `sizeof(ua)` instead. This seems to work fine for this.

``` patch
--- vmware-sercons/sercons.c    2016-10-09 13:46:48.461275841 +0200
+++ vmware-sercons_/sercons.c   2016-10-09 13:43:32.027692466 +0200
@@ -191,10 +191,9 @@

        bzero(&ua, sizeof (ua));
        ua.sun_family = AF_UNIX;
-       ua.sun_len = sizeof (ua);
        strcpy(ua.sun_path, path);

-       if (connect(fd, (struct sockaddr *)&ua, ua.sun_len) == -1) {
+       if (connect(fd, (struct sockaddr *)&ua, sizeof(ua)) == -1) {
                (void) close(fd);
                return (-1);
        }
```

**Comment by sjorge** [comment]
Been testing this on kvm vm.console sockets, the support for OpenBSD/FreeBSD using socat is poor.

e.g. vi/vim/nano are mostly broken when using socat. 

Using sercons makes them think a vt220, this makes PAGER and editors behave as they should.
More importantly passwords now seem to be properly hidden.

```
[root@mantle ~]# ./sercons -l out.log /zones/47643fc2-e06b-4913-9cab-74a0b7833514/root/tmp/vm.console

 * Connected.  Escape sequence is <CR>#.


OpenBSD/amd64 (exosphere.acheron.be) (tty00)

login: sjorge
Password:
Last login: Sun Oct  9 13:54:21 on tty00


    _                        )      ((   ))     (
   (@)                      /|\      ))_((     /|\                        _
   |-|                     / | \    (/\|/\)   / | \                      (@)
   | | -------------------/--|-voV---\`|'/--Vov-|--\---------------------|-|
   |-| exosphere.acheron.be    '^`   (o o)  '^`       OPENBSD60JS20FW14P | |
   | |                               `\Y/'                               |-|
   |-|                                                                   | |
   | |                                                                   |-|
   |-|                                                                   | |
   | |                  UNAUTHORIZED ACCESS PROHIBITED                   |-|
   |_|___________________________________________________________________| |
   (@)              l   /\ /         ( (       \ /\   l                `\|-|
                    l /   V           \ \       V   \ l                  (@)
                    l/                _) )_          \l
                                      `\ /'
                                        `



[exosphere :: sjorge][~]
[!]$ env | grep TERM ; exit
TERM=vt220

 * Escape Character Received.
```

Would love to see this replace socat or be available as an alternative to it in SmartOS.

**Comment by jclulow** [comment]
This should be fixed in 43e3b31.  I tested it on SmartOS against the `/tmp/vm.console` socket in a KVM zone.

**Comment by sjorge** [comment]
Just compared the new version, I can confirm it works fine on SmartOS now for vm.console socksts!
